src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -594,13 +591,23 @@ public void observe(Operation operation) {
   @Override
   public void addWorker(ShardWorker shardWorker) throws IOException {
     String json = JsonFormat.printer().print(shardWorker);
+    long firstRegisteredSeconds = MILLISECONDS.toSeconds(shardWorker.getFirstRegisteredAt());
+    int firstRegisteredNanos =
+        (int)
+            (MILLISECONDS.toNanos(shardWorker.getFirstRegisteredAt())
+                - SECONDS.toNanos(firstRegisteredSeconds));
+    Timestamp registrationTime =
+        Timestamp.newBuilder()
+            .setSeconds(firstRegisteredSeconds)
+            .setNanos(firstRegisteredNanos)
+            .build();
```

Comment:
Refactor this into a single function that takes a ms count and returns a timestamp. Use the var name `effectiveAt`, or even better, don't retain this in a variable, and put it directly into the WorkerChange.